### ISSUE TITLE
False positive from unnecessary_this

### ISSUE THREAD

**Issue by bwilkerson** [dart-lang/linter#1915]
The following code triggers an `unnecessary_this` lint, but removing the `this` isn't valid:
```
class C {
  int i = 3;
  void loop() {
    for (int i = this.i; i < 1; i++) { // false positive
      print(i);
    }
  }
}
```

**Comment by camsteffen** [comment]
This is fixed in 4bfa66e1acd286c778b37b57567f1c1368d4be6a

**Comment by pq** [comment]
Good catch.  Thanks @camsteffen.
